src/DynamicMvvm.Abstractions/ViewModel/IViewModel.Extensions.Properties.cs

```diff
@@ -65,7 +63,40 @@ public static partial class IViewModelExtensions
 			this IViewModel viewModel,
 			T initialValue = default,
 			[CallerMemberName] string name = null
-		) => viewModel.Get<T>(viewModel.GetOrCreateDynamicProperty(name, n => viewModel.GetDynamicPropertyFactory().Create(n, initialValue, viewModel)));
+		)
+		{
+			// We don't use GetOrCreateDynamicProperty internally to avoid the performance costs of the lambda and closure.
+			if (viewModel.IsDisposed)
```

Comment:
there seems to be duplicated code. I'm wondering if there could be a way to refactor the repetitive part into a separate method

Reply:
That's the cost of the optimization.
Each branch of each method as a return statement, so it cannot really be refactored too much.